assembly/Hash.ts

```diff
@@ -0,0 +1,43 @@
+import { Codec } from "./Codec";
+import { Bytes } from './utils/Bytes';
+
+export class Hash extends Array<u8> implements Codec {
+
+    constructor() {
+        super(32);
+    }
+
+    // Encodes
+    public toU8a (): u8[] {
+        const result: u8[] = new Array<u8>(this.encodedLength());
+        Bytes.copyFromPosition(this, result);
+
+        return result;
+    }
+
+    public toString (): string {
+        return "0x" + this.join('');
+    }
+
+    static bytesToHash (bytes: u8[]): Hash {
```

Comment:
I think that `bytesToHash` is the same function as `fromU8a`..
You have an input -> array of bytes and you return Hash in both cases